sphinxcontrib/chapeldomain/__init__.py

```diff
@@ -56,14 +56,62 @@
            (?:[+*/!~%<>=&^|\-:]+)                  #  or operator name
           )  \s*
           (?:\((.*?)\))?                           # opt: arguments
-          (\s+(?:const\s)? (?:\w+?)|               #  or return intent
-           \s* : \s* (?:[^:]+?)|                   #  or return type
-           \s+(?:const\s)? \w+\s* : \s* (?:[^:]+?) #  or return intent and type
+          (\s+(?:const\s)? (?:\w+?)                # opt: return intent
+          )?
+          (\s*:\s*(?:[^:]+?)                       # opt: return type
+          )?
+          (\s+throws                               # opt: throws
           )?
           (\s+where\s+.*                           # Where clause
           )?
           $""", re.VERBOSE)
 
+
+def match_chpl_sig_pattern(sig: str):
+    """
+    Match a Chapel signature against the regex pattern defined in
+    chpl_sig_pattern. chpl_sig_pattern cannot be used directly because we need
+    to fix some things up
```

Comment:
I'd give an example of something that needs to be adjusted for here.  My brain reading this comment first went to "oh no, is there a bug we're sweeping under the rug?!" when I think what you meant was just "we need to do some special interpretation here"